packages/databyss-ui/components/PDFDropZone/PDFDropZoneManager.js

```diff
@@ -0,0 +1,244 @@
+/* eslint-disable no-plusplus */
+
+import React, { useState, useEffect } from 'react'
+
+import ObjectId from 'bson-objectid'
+
+import { BlockType } from '@databyss-org/services/interfaces'
+import { useEditorContext } from '@databyss-org/editor/state/EditorProvider'
+import * as services from '@databyss-org/services/pdf'
+
+import { useNavigationContext } from '../../components/Navigation/NavigationProvider'
+import { View } from '../../primitives'
+import InfoModal from '../../modules/Modals/InfoModal'
+import styled from '../../primitives/styled'
+
+import DashedArea from './DashedArea'
+
+// constants
+const ACCEPTABLE_KINDS = ['file']
+const ACCEPTABLE_TYPES = ['application/pdf']
+
+// styled components
+const viewStyles = () => ({
+  position: 'absolute',
+  bottom: '4%',
+  height: '100%',
+  left: '50%',
+  marginLeft: '-48%',
+  overflow: 'hidden',
+  pointerEvents: 'none',
+  width: '96%',
+  zIndex: '10',
+})
+
+const StyledView = styled(View, viewStyles)
+
+// methods
+const isAcceptableFile = item =>
+  ACCEPTABLE_KINDS.includes(item.kind) && ACCEPTABLE_TYPES.includes(item.type)
+
+// component
+const PDFDropZoneManager = () => {
+  const editorContext = useEditorContext()
+
+  const { showModal } = useNavigationContext()
+
+  const [isDropAreaVisible, setDropAreaVisibility] = useState(false)
+
+  const [isParsing, setParsing] = useState(false)
+  const [hasParsed, setParsed] = useState(false)
+
+  // utils
+  const getFileToProcess = event => {
+    const filesToProcess = []
+
+    if (event.dataTransfer.items) {
+      // Use DataTransferItemList interface to access the file(s)
+      const numItems = event.dataTransfer.items.length
+      for (let i = 0; i < numItems; i++) {
+        const item = event.dataTransfer.items[i]
+        if (isAcceptableFile(item)) {
+          filesToProcess.push(item.getAsFile())
+        }
+      }
+    } else {
+      // Use DataTransfer interface to access the file(s)
+      const numItems = event.dataTransfer.files.length
+      for (let i = 0; i < numItems; i++) {
+        const item = event.dataTransfer.files[i]
+        if (isAcceptableFile(item)) {
+          filesToProcess.push(item)
+        }
+      }
+    }
+
+    if (filesToProcess.length === 0) {
+      return null
+    }
+
+    const file = filesToProcess[0]
+
+    // TODO: handle multiple files at once?
+    if (filesToProcess.length > 1) {
+      console.warn(
+        `Multiple files are not handled at this time. ` +
+          `"${file.name}" will be parsed.`
+      )
+    }
+
+    return file
+  }
+
+  const toDatabyssBlocks = (fileName, annotations) => {
+    // create response object with source as first item
+    const response = [
+      {
+        _id: new ObjectId().toHexString(),
```

Comment:
Now that I think about it, we re-generate all `_id`s on an `insert` operation for ENTRY blocks, so let's try leaving this out for clarity.

Reply:
So I simply remove that  prop?